File: solutions/1503. Last Moment Before All Ants Fall Out of a Plank/1503.py
# Time complexity: O(n + m)
# Space complexity: O(1)
class Solution:
    def getLastMoment(self, n: int, left: List[int], right: List[int]) -> int:
        ans = 0

        for num in left:
            ans = max(ans, num)

        for num in right:
            ans = max(ans, n - num)

        return ans

File: solutions/1503. Last Moment Before All Ants Fall Out of a Plank/1503.cs
// Time complexity: O(n + m)
// Space complexity: O(1)
public class Solution {
    public int GetLastMoment(int n, int[] left, int[] right) {
        int ans = 0;

        foreach (int num in left) {
            ans = Math.Max(ans, num);
        }
        
        foreach (int num in right) {
            ans = Math.Max(ans, n - num);
        }
        
        return ans;
    }
}